source/vibe/core/sync.d

```diff
@@ -122,26 +122,30 @@ class TaskMutex : core.sync.mutex.Mutex {
 
 	override @trusted void lock()
 	{
-		if (tryLock()) return;
-		debug assert(m_owner == Task() || m_owner != Task.getThis(), "Recursive mutex lock.");
-		atomicOp!"+="(m_waiters, 1);
-		version(MutexPrint) writefln("mutex %s wait %s", cast(void*)this, atomicLoad(m_waiters));
-		scope(exit) atomicOp!"-="(m_waiters, 1);
-		auto ecnt = m_signal.emitCount();
-		while (!tryLock()) ecnt = m_signal.wait(ecnt);
+		try {
+			if (tryLock()) return;
+			debug assert(m_owner == Task() || m_owner != Task.getThis(), "Recursive mutex lock.");
+			atomicOp!"+="(m_waiters, 1);
+			version(MutexPrint) writefln("mutex %s wait %s", cast(void*)this, atomicLoad(m_waiters));
+			scope(exit) atomicOp!"-="(m_waiters, 1);
+			auto ecnt = m_signal.emitCount();
+			while (!tryLock()) ecnt = m_signal.wait(ecnt);
+		} catch {}
```

Comment:
I didn't want to change the `ManualEvent` interface to nothrow. It's going to create a huge mess, and I only wanted it to run for now. Maybe I could turn any exception into a throwable through an assert in the catch clause?